lib/grinder_tools.dart

```diff
@@ -84,53 +85,57 @@ void runProcess(String executable,
       environment: environment);
 
   if (!quiet) {
-    if (result.stdout != null && !result.stdout.isEmpty) {
+    if (result.stdout != null && result.stdout.isNotEmpty) {
       log(result.stdout.trim());
     }
   }
 
-  if (result.stderr != null && !result.stderr.isEmpty) {
+  if (result.stderr != null && result.stderr.isNotEmpty) {
     log(result.stderr);
   }
 
   if (result.exitCode != 0) {
-    throw new GrinderException(
-        "${executable} failed with a return code of ${result.exitCode}");
+    throw new ProcessException(executable, result.exitCode, result.stderr);
```

Comment:
sgtm.  At that point we're only missing the `pid` from `ProcessResult`, so should we just attach one of those to the exception, or keep the individual fields?  I'd say keep the individual fields.

Also let me know if you don't like the name `ProcessException`.  I initially had `NonZeroExitCodeException` which is the most accurate, but a bit long.  Could also be `ProcessRunException` or given #161 maybe `RunException`.